### PR TITLE
Improve React Hooks call

### LINKED ISSUE #11881: Opt into React Hooks formatting
**Prettier 2.5.0**
[Playground link][1]

**Options:**

```json
{
  "arrowParens": "always",
  "bracketSameLine": false,
  "bracketSpacing": true,
  "embeddedLanguageFormatting": "auto",
  "htmlWhitespaceSensitivity": "css",
  "insertPragma": false,
  "jsxSingleQuote": true,
  "parser": "typescript",
  "printWidth": 80,
  "proseWrap": "preserve",
  "quoteProps": "as-needed",
  "requirePragma": false,
  "semi": false,
  "singleQuote": true,
  "tabWidth": 2,
  "trailingComma": "none",
  "useTabs": false,
  "vueIndentScriptAndStyle": false
}
```

**Input:**

```tsx
const closePortal = useCallback((e: any): void => {
  if (isServer) return

  const event = createCustomEvent(e)

  if (isOpenRef.current && typeof onClose === 'function') onClose(event)
  if (isOpenRef.current) open(false)
}, [createCustomEvent, isServer, onClose, open])
```

**Output:**

```tsx
const closePortal = useCallback(
 (e: any): void => {
   if (isServer) return

   const event = createCustomEvent(e)

   if (isOpenRef.current && typeof onClose === 'function') onClose(event)
   if (isOpenRef.current) open(false)
 },
 [createCustomEvent, isServer, onClose, open]
)
```

**Expected behavior:**

The output should match the input.

**Additional context:**

From what I understand, Prettier has special logic for formatting React Hooks, but it doesn't seem to apply to `useCallback` in some cases. I don't have this formatting problem with `useEffect` (intended, I know), `useMemo`, or `useCallback` if using an empty dependency list.

I find the output to be less readable than the input and would like to skip this particular type formatting in regard to `useCallback`. Using `// prettier-ignore` where applicable is **not** a favorable solution (merely a hack where I'm concerned).

I've searched for the repo for related issues and Stack Overflow for possible solutions, but please let me know if an option or configuration like this already exists and I've just been overlooking it!

[1]: https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBLAtgBwgJxgAhgE9M59h8B5HAOQFcAbBgGnwAUBDGACwDUOGdMgF98AMxwR0+AOQABMQzgAPALQATOADc4DCJnQIYAehh0YqBgGcZAHSgZseQiTIUAshDpW4AUR2wrADKRLDccBZg-kb4ohJSsjhwHGAwdg5YuAQU3nAAwgIMAEYpANasub5iYnCpFT7ucOgQ9XAASnBirUEwXCLiktIySSlp9o5Z5OKoUOoAIpTuNBCasQMJw8mpGlLpE865QUFt60O5qlZWOHuZzsSk5Pb4+HneMFLRgU8vbx8BMAAJDizJQ4Kz4DjgoEguBg5jfNhZAQdMT2OKDWQAOmM5ycvQYmPucBs41u2W+rys73Qn0BwPUoIhUPpoPhUGeAFUfIi8AI2ZzuUiGJRMBZoFY0acsTifKo8QJCa4SVB7MYAFRqp5q-AKSxkAEQCClcGqfC5Hn4rU65rqRhkOioYzcQ3GmVwC0CLXGeyqjVWuTwLAMPrsfCmgEAFXcABl8LomjEifgIGJ2EL-YHMMH4PgI2H8JGY3GlIZYC4Him07yGIQcKgAOb12Fa-2YDg4DjSYBc91CkViqBWUQAbX0A6sAF18wajcnRahxa3253yIXo74S0YR2OF4PMUUZuoIxAALzqCBgOilmD7lZEKemj01ri9MDca9LjtdoqGpTA7fzuKmJgHoPiUFAvhWGAJ4wDgQgPi8oFkPKNaoKmAAGxJgOh+ClHARD4Kg4LcKgMCfiuwA-hAf5QAB47AUh4GUOYMFwXACF5Eh+Aocm7JeDAVioKsIGoGApTkd+v7JLR+CjoBe5ESKCAnmIAg+AhQQRPgmiqYwBDoYppBQDhVJ9BJq4srCdG7lYmLQJxEDqfmBRMCUYmEBA0xJPgADu4TsjxRH4CBjlwOo5nANCDJWbJO5AdASlQBxhRuaUHleWQfkINxQqEeC+gIGFEVRaC1nxVAT6caJpTJa5ZTpWIqDedA+DoU+OEiWJxVRuum6wGVe4oQhG4JmW7ziHQUCpLuTKte13UxiN14DbZsENk2ODDX1BDjWtjawnNQo4QVDhQPW-pJGYODst2grVqIM5paZ8BevYYiTdNLXmkKAA8CJxso8CzFCPVLTEAA++BQIwNYngWoPbfgkPQ0w-K5gDQPqCDi2I8jMP4HDa5g2WeNMPYAB8AAUwDfAeszHmjIVgRBUEE7WQiM4xUDMQQcOwRz3yGQgSDaZ0HB6QAklYiVs6p1hwGj9lIYr3NGSrlUMNVaMoSL6g8XD56XtewEjPAxMwJTMjqKgWgyAAlGje0bbrTuwmjmIe9wllgmiIs9k+-Y2WzwDCHbft3fiP1sKwEasHhRCVkSlZezCYLk487L4MYxj4EEdCYHi+A+DgOg4EXQlkEksywjM52Z5Ag7ZHlABCkg+cXrBEZpJcHaIcOHMclN2z6mfZ-gEsOAQJ1F70Ob0vgSQ1FXYBwN8DdUrJQtQKwPgwFLiVTv3Pg9H0P1UTRVM6eLDB79LRnDyqo853kpuV50sIICv6Vb+lmBJP8M8BBkC0AIIQEp67igIFvFEbNcgojPlJYEVMt4P2+OqTUmd8Dag5JgdQfRwQ8DIAZO+CAcLz2IYlFEJlZ7EnLugPSwI4BeCsAwIgmJvhYI4dqOQbYvzkHPtJUQz0yCmhoHAHygCXqYO4ZdOg11yBaAgEJUQyweC118v5YKUgswRFXtI70EDG5zmykffIKUyiU0psIkWAjgRh3wIooSBN0400wUXCI+8jKWOgZ0YCciq680kXAO2D9njCFYMOCcqCn4vFfgvd+y8yDjR4vPHirscBr0gTlassCfDwLYEjKGMNkFWFbhAduB0AD8+BKZ61ypCM0UBShQHKeyBpbA7b4BFijBgoStFGPSbk9onQfp5lJgwKmtTBkNMms01ps0IydPBuM6Jzwx44LwTmHii8P5TT0c8So1RagWyHs4jOzxnhoRqURMpFSy4ADJ7n4AAIQoT8TgAJnS3mXg+TEA2KTwRsG+OE2SdTqyd1KW3YuUSR5rJzgAcS0gsdwxZRo7U8i+FI3Bsn4g8pkoxzRJqBNyI0ZoP0ia40aTpGYYVJmdJPC4jhWcc5zE8i0tRZ1CKpirpoOsnKfClwuBXJlVzKZd1hKXTpsjrqwouWPYkmtYAaCIhwIoShVBQBUDAVQiq36jGMC01QjVZg7HQBqlY+yLnSvZJYumR5PKPOmLMZFyxNCUztceEJSNIaGyvEYW86giCdIaRStFwKIkepaHlbupcYWPzWX6aRsTkjwHBBwcgchFVpUpNSWkwh2HSK4TqXhFFgREFECI-A-BBBJM8g3UuUDYCeXQjmv4Rh0JFrkNa8graaT-BUeIn4VI22wC4QY5468CBgFfr22kQyXLFAsZTOAIsy32Nnf8KOMd04MvORc0VzzgnxKujdOM-wRYKxxXyWs61e6zQ3UYLduZyayueHATEpAwREQII699n7BJUiHky992g-lxmAx+3KcMUIQcGXzOs+0Mnxv3amSmzz0nvICfgX9mGjARnbE2H9Tz30-ICfhnAhGXkngNheP1sBOmuIuRcjDpGwMkf8XhgjEQmXCCZd2y1ILImrOZfgRFU7fjSFA2WFO0UfaGI3lJukqdwSmNJRAcl3srB0rOYx5jrh8C+FgHWWhcNhzDnjpWEqsIrCsE4DwatQgfodAAI4OiSOoDTynybkwnBOV9-SFNGdQCZmplAigACtjmYiMMZqwlMZOgisMG8Ehm1rEjtpidAHBMCWOHFATsl6xBJXpenYcTLnj5cMGjJjy7-iVJFiEMIERRK0h+mW1gtJyYlb3Uxy5qHxU9xwFKiIcjkO9fEFASm06U35Ak7SWrRhkvsGXIYeAYIfpJ1TEV8mw4AAMcbxu8bcXGvjI35Fhci6kTE8QaRBeJMuu7SXw2yWm30B9gR8AJesxCmNsITuj0TRc7UDkfAEPCMFDjZY3lcM7SWrsZaK3OS4opzt3bgCOPUCoiAHL6waOypAIMuix34o3kzXsOTTELtSpYldEIoBBpFhjnTIr+tWF+0N49o3-OTrPWB178B3sW2Cf50VPixC4bLI6zbvEQdkGo3DGQ70poDntjLpCC36Ms+uSQqAKIJcwE6SdSmcsfB9ME-z2bw6+1GB+xKt2avQqsBOv9hNGCgdUCMmDsgrGodCgLUDotAB1TRmEcJBXQjQCW0Z0KsEId5IK6aE-4FsCAHycAmrqAXikb30BrYDnYSAGHSbjzcX-omEiNmM0+CSQhja+vRD0I3kUMgwJWqhuvOhTE+AACC7JYSSCQ+7nylgazN8INwNu7IrnoRY5D3m1GilME70X93PCVvkAR3GfMMsUdJq7Wdm6GOsc47x+yAnOipGr54FC8gvgPm4FEBLVMKe08Z6z1-Buefdwp7yoRKAii8Jwp9EScp4jInx51zExIadV16d7Emdd1dM+ttd2dhsT1ucslFM2YLdBdl1hMkDKZZ9fkyx5dF9ekesLld4IwMAmFzBLFusjd+ZglWBds8DmMJ9ykoZB078B9LZu91BM9010JF4cJdpwd4xrx8AE46AIR49YBYRRh1Fh8eBkwy42xLg6dWpFMRDPITonxcCVCPJGwlA9Dgl7YeMRdUNpcEojICYF9FcPoVdDdVYEANcDcmUGC2IzcIlsC5t-hbdBsndnDt5jFitZV0Ei1Zd8oy4TovcIciCCBodC0984cN96dEdTRjwjCyBd9V80cj98BVESJOUsoz9tElBL9ng1Rx0At0UsjwDKdIDShoC6cGcHElFM8EDTsT08pKFfEfcCBqkycTDOkRZdChRcDnthwhihRWAQinwXcs5AdKjEJQpYj+ir0awSjx8iEsIcJ45f9OpAD-dKjYd19gAABpfCH8dsdQPNXnMsU0S4oga4nATPHI5Y-fbo9Hdo4-Io3HLY8-coy1LBaonnL7SCMAbLOXM0HwKnJdRTEWJ4l424-4OA9o5nNxUVMncCCE7DYjf4TEfYkgmQCEqE1XaY6sVwrwl7LmCE1gCk-Ef7NBJYrBFY0HbYuIrDHiLY9NXIMuQ45McwQSVYSsUicEH8SaLGY41k04vhYATwSofte4ggU0BUnwZU1HA-BRH4go7HP40-LRQnCokEkAz7SyHmYU-ITWdyBouqKA1wkWNUvwVExndEzotxcItxVkvIcIMScEafXoCjCIfXDqaAXoGYcETCAkwMwjZfL07UcrbhFI4AagegJgKOIUFEcmEcGMiIDiMMjgGlMucQowRMnUNHWxGSGfNiMPVMBucMzVdQVgdCE3OAY6OPYfHwRM6oidLJeswswcWkcjSjOGAgrjGAerKgWgGGDM6sLM+xSsjE8bbtXMic-XSpYCAsiM1wwkcc2advRbfASpapVs8w9kJlMeH02oNKK5TA1c3-CMiuDYgwtJWvZsNxHnfsmlTPNmMrL0-pBsqwIc8cymAFKsfEFEB2crACgcoC-4YciIAgt8sudpTMzoPpZ4CcTEKwKQOAKbLc9kXdZ5Z5L8qAaJC8llNlPU9RK5AUu8vcoKJQoo583AZ89JLXfC2AAcsKb1GpTAhpNdfccwd4dkZ5BfXbXi55MXfXVArnMbETWXFiviIUp8gU28-LNVIqTE1DbE7mcwTpBkgQKkyY3SnmekpCeY8igHN3ZYqzcEHiQ42yItWUiiJ0u4ytJ0jUvfPInUwoxQzRQEonYA+TAgL7J01lHyQimEsxO0po5dR05hZ0xbV0pxd0pjUXNnO3MuZZF5UCv3dYiM3oPZSzBGNFL1a1JlJlHnTAuGHwq3ebYXD8vsgini6DfKufTcri7cxTXcoMggENUq68VgrElqzPKXVwSsaADWaqWwhXJXT6KAVXKaoUKqKAxTDCs0mEC0iuVapoxqsJbwmdXwm3Ta6KbazQXa-w0uQI6asSBYtHJlD2aLf4KzOEKqiymYkVHXEWaS-o6rOYz6txFCf694OowGpjdizOUQBpf2X6aOXMOOfCROCa1ML7NONEEeFQQuK+PSaKp8ewEAZgEAOKQcZAUAdsNuTgKuKwZAEAAQHyDgIgGmomooDsMSCIIIAraMGlZAVslmtmvCGAIINsMAWuZARgompoZvfgsKaMYEesOgDgJsAAMVwCyxgAsDOlpvFneEJpAG4BgHQAYGD1ImJBFrgE0kHFIhtlIiIFprAEuD1ojNhBgDYA7HrCy15rUgVhAHCysGUCCFriUAAEU6Bsc4Bxa2IibeFi5aaiQoI6xRQ9a-4ZgYBA8hIeBkAAAOXbaOyQHwQPDsTAWmv+YkO3PW1zcOt2-QGmlASEDVOAMKMKPWpIVzJqd0d2z2pAPmkAHwdAVAL2+WImwSM6EOsO+ASOjmEAXoIodO9QTOpAAAJiJtgkLMVXrDyCkC7pABaU1T1tyHwyKFrp7q0CEAnk0FgCCGnVQFFF73UB6CICUEHp8GEGECAA

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Description

Fixes #11881

## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by sosukesuzuki** [comment]
I don't know ideally formatting. But inconsistently formatting that depends the number of args and deps is not good:

**Prettier 2.5.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEB6VACAhgJwOYDO2UAJhiXAA5G45YCeAOlJFATBgGYQQYC8GAK4E4AYSwAbCQCMsYANYAKRZVxYAtgQCU-AHwZgzDBnTleBCOrgwAFgEsoeZgF8ANBgDaFSgEYAuloA3MzMpiSUcDQ4dEws0OxcPPxCIuJSsgrKOnz6hlDGYeaW1vaOLu5eVP5BIVCmuITMrAmyOMnCYpIyckoqaprZuUYmmCRFVrYOTlBungHBUCCuIBCUMHbxyKC0EADuAAq4CATIIJK7DCfL0nQK1gDKGnAAMg5wyJySIte38g+qYCmyBgOEEcGWcHU0jgJAoJGeWEcgiweDgADEIDh1FgYOtHKcsIIYBAliAbDB1BIAOr2eAEAFwe7HOzrABuLPopzABCuIAcIhwMH2dDw2I+X3BIAAVgQAB73KYSOAARUEEHg4ok3xAqhwAtOsmhElJlBwDhgVLsJFsyAAHAAGZamiAiKl0SinU2ROA4VnvZYARzV8GFqxOKCwBAAtFA4DCYaScHAg3Yk8KUWKkJ8tZKROo7MDQbnFSrg+8sxLljAsNJLdabMgAExVuh2CRTUSWTMgSIAVlJHQAKjXw9ntaywQBJUgIGD3MBmtYAQVI9xg9CVmpEzmcQA)

```sh
--parser babel
```

**Input:**

```jsx
// args and deps array
const foo = useCallback((params) => {
  // do something
}, [dep1]);

// dpes array
const foo = useCallback(() => {
  // do something
}, [dep1]);

// args
const bar = useCallback((params) => {
  // do something
}, []);

```

**Output:**

```jsx
// args and deps array
const foo = useCallback(
  (params) => {
    // do something
  },
  [dep1]
);

// dpes array
const foo = useCallback(() => {
  // do something
}, [dep1]);

// args
const bar = useCallback((params) => {
  // do something
}, []);

```

File: src/language-js/print/call-arguments.js [hunk 1]
```diff
@@ -271,11 +269,39 @@ function shouldGroupFirstArg(args) {
...
+ * @returns {(string | undefined)}
+ */
+function getCallName(callExpression) {
+  const { callee } = callExpression;
+  switch (callee.type) {
+    case "Identifier":
+      return callee.name;
+    case "MemberExpression": {
+      const { property } = callee;
+      if (callee.property.type === "Identifier") {
```
Comment (B2o5T) [inline src/language-js/print/call-arguments.js:286]: Use property.type directly as you already destructured this

**Comment by raphael-nazirullah** [comment]
When is this PR going to be merged into prettier? ☹️

**Comment by timkolotov** [comment]
Can we do something to get it merged sooner?

**Comment by sosukesuzuki** [comment]
This PR is too old, I'll re-open as a new PR

**Pull request closed without merging**